File: app/src/main/java/com/example/mobv/Adapter/GifAdapter.java
package com.example.mobv.Adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.mobv.MessageActivity;
import com.example.mobv.Model.GifResource;
import com.example.mobv.R;
import com.example.mobv.api.responses.Contact;

import java.util.List;

public class GifAdapter extends RecyclerView.Adapter<GifAdapter.ViewHolder> {

    private Context mContext;
    private List<GifResource> gifs;

    public GifAdapter(Context mContext, List<GifResource> mGifs) {
        this.gifs = mGifs;
        this.mContext = mContext;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.gif_item, parent, false);

        return new GifAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        final GifResource user = gifs.get(position);

//        val imageView = findViewById<ImageView>(R.id.webView)
//        Glide.with(this).load("https://media0.giphy.com/media/65ODCwM00NVmEyLsX3/giphy.gif?cid=3f233d15475808b0a42b36e882b128e979e49f69b505ffa9&rid=giphy.gif").into(imageView)

        Glide.with(mContext).load("https://media0.giphy.com/media/65ODCwM00NVmEyLsX3/giphy.gif?cid=3f233d15475808b0a42b36e882b128e979e49f69b505ffa9&rid=giphy.gif").into(holder.gif);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Postnut gif
            }
        });

    }

    @Override
    public int getItemCount() {
        return gifs.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{

        public ImageView gif;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            gif = itemView.findViewById(R.id.gif);
        }
    }

}

File: app/src/main/java/com/example/mobv/adapter/GifAdapter.kt
package com.example.mobv.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.mobv.model.GifResource
import com.example.mobv.R
import com.example.mobv.databinding.GifItemBinding

class GifAdapter(
    private val mContext: Context,
    private var gifs: List<GifResource>,
    private val listener: GifListener

) : RecyclerView.Adapter<GifAdapter.ViewHolder>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        return ViewHolder.from(parent)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        val gif = gifs[position]
        holder.bind(gif, listener)

        Glide.with(mContext)
            .load(gif.images.fixed_height.url)
            .into(holder.gif)

    }

    override fun getItemCount(): Int {
        return gifs.size
    }

    fun update(gifs: List<GifResource>) {
        this.gifs = gifs
        notifyDataSetChanged()
    }

    class ViewHolder(val binding: GifItemBinding) :
        RecyclerView.ViewHolder(binding.root) {
        var gif: ImageView = itemView.findViewById(R.id.gif)

        fun bind(item: GifResource, listener: GifListener) {
            binding.gifResourceModel = item
            binding.clickListener = listener
            binding.executePendingBindings()
        }

        companion object {
            fun from(parent: ViewGroup): ViewHolder {
                val layoutInflater = LayoutInflater.from(parent.context)
                val binding = GifItemBinding.inflate(layoutInflater, parent, false)
                return ViewHolder(binding)
            }
        }

    }

    class GifListener(val listener: (gif: GifResource) -> Unit) {

        fun onClick(gif: GifResource) = listener(gif)

    }

}